### DISCUSSION TITLE
Make a better Article/Related Items Module for J4

### DISCUSSION THREAD

**Discussion by V1nc3nt79** [joomla/joomla-cms#36652] [2. Cold Features]
Hello,
First I want to say that I love Joomla and I'm very grateful for the work you are doing.

I'm using Joomla for 14 years now but I'm very sad/disapointed to see that there's still not a proper/good Article/Related Items Module in Joomla 4. 
**I think that's a major and the minimum feature a CMS must have.**

For the last years I'm using a module called DisplayNews by BK (https://extensions.joomla.org/extension/news-display/articles-display/displaynews-by-bk/). It will be great to make something similar and fully integrated to the system.

Second thought : It will be great if **modules names could link to a page**. For the moment, i'm using a hack to do this. But I think it's a must have feature. For example : you've got a module presenting last news but there's no way to link to the news page presenting all the news.

Thank you to take time to read, i write this to make Joomla the best CMS.

**Comment by brianteeman** [comment]
1. If the module exists to do what you want then why should it be duplicated into core? Remember what is essential for you is not for everyone
2. Can you explain a bit more please. Maybe with images

**Comment by V1nc3nt79** [comment]
1 - II agree with you but for the moment the Category Article modules and related module are very basics, not really usables, and have a lack of functionalities. For example : it's not possible to display images (intro or full). In the tag selection, it's not possible to choose to display according to all tags or just one of them, etc...

2 - Imagine you have a module displaying last events :  
Last Events
- Event 1
- Event 2
 -..
It could be nice if it was possible to click to the name of the modules "Last Events" to access all the events. (but same principle for articles/categories/..)
I think all modules name should have the abilities to be linked to a page/section of the site.

**Comment by brianteeman** [comment]
Ah so you want the title of the module to be a link?

**Reply by V1nc3nt79** [reply]
yes! (but first i'd like a better article module!) :)

**Reply by brianteeman** [reply]
That is why joomla is extendable

**Reply by V1nc3nt79** [reply]
because of the lack of basic and functional feature ?

**Reply by brianteeman** [reply]
No, because everyone has different needs

**Comment by V1nc3nt79** [comment]
Thank you to have made a proper article module in Joomla 5.2 !

**Reply by MrSuddenJoy** [reply]
@V1nc3nt79 is this an irony? Definitely sounds like it. Cause article module in 5.2 is (sad) joke.

**Reply by Hackwar** [reply]
You are welcome to further improve it and provide PRs.

**Reply by V1nc3nt79** [reply]
It's not irony; Joomla is finally taking users' needs into account. Why would that be a joke? The module is good and could certainly become even better.

**Reply by MrSuddenJoy** [reply]
@V1nc3nt79 if so, than read this: articles we publish on our site are quite lengthy and its not rare (9/10 times) article module crashes when we hit `Save` button.

Whats worth noting is:
* this does not happen when article is < 10k chars,
* this does not happen when we save & publish article.

**Reply by V1nc3nt79** [reply]
Report the bug!

**Reply by Hackwar** [reply]
So, after you saved, the module in the frontend crashes because your article is so large?

**Reply by Hackwar** [reply]
And what is the error message that you are getting?

**Comment by MrSuddenJoy** [comment]
> Report the bug!

@V1nc3nt79 will do shortly 👍🏻

> frontend crashes because your article is so large?
 
@Hackwar seems to be true 👍🏻

> And what is the error message that you are getting?

@Hackwar will screen grab it once it appears.......

Maybe naive question @Hackwar, but would it be possible it is NOT Joomla's responsibility? I mean the root cause beeing on our side of things (infra for example)?

**Reply by V1nc3nt79** [reply]
@Hackwar I noticed that there are indeed issues with module saving at the moment; this happens to me with the Custom Content module.
